dlt/common/configuration/container.py

```diff
@@ -87,10 +87,14 @@ def _thread_context(
             context = self.main_context
         else:
             # thread pool names used in dlt contain originating thread id. use this id over pool id
-            if m := re.match(r"dlt-pool-(\d+)-", threading.currentThread().getName()):
-                thread_id = int(m.group(1))
-            else:
-                thread_id = threading.get_ident()
+            # print(threading.currentThread().getName())
```

Comment:
I'm not quite sure how this is supposed to work, in any case it will not work to inject multiple configcontexts of the same type in different threads if I keep this in. lmk.